test/integration/webhook_token_auth_test.go

```diff
@@ -0,0 +1,207 @@
+package integration
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"testing"
+
+	kauthn "k8s.io/api/authentication/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	restclient "k8s.io/client-go/rest"
+	kclientcmd "k8s.io/client-go/tools/clientcmd"
+	kclientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/apis/config"
+	"github.com/openshift/origin/pkg/oc/cli/cmd"
+	userclient "github.com/openshift/origin/pkg/user/generated/internalclientset/typed/user/internalversion"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+// TestWebhookTokenAuthn checks Tokens directly against an external
+// authenticator
+func TestWebhookTokenAuthn(t *testing.T) {
+	authToken := "Anything-goes!"
+
+	expectedTokenPost := kauthn.TokenReview{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "authentication.k8s.io/v1beta1",
+			Kind:       "TokenReview",
+		},
+		Spec: kauthn.TokenReviewSpec{Token: authToken},
+	}
+
+	tokenResponse := kauthn.TokenReview{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "authentication.k8s.io/v1beta1",
+			Kind:       "TokenReview",
+		},
+		Status: kauthn.TokenReviewStatus{
+			Authenticated: true,
+			User: kauthn.UserInfo{
```

Comment:
Send and check all user fields.

Reply:
UID and Extras do not carry over to user Info, so I am adding a check froor Groups only.

Reply:
Check scopes by doing a call that will fail due to the scope.

Reply:
> UID and Extras do not carry over to user Info

Really? They should, right?

Reply:
> Really? They should, right?

Pretty sure @simo5 meant "UID and Extras do not carry over to `get user ~`"

Reply:
what I mean is the WhoAmI call returns a structure without the UID and there are no Extra fields at all.
@enj please suggest a call and a scope that would satisfy your idea of a check.